package.json

```diff
@@ -18,19 +18,20 @@
   "types": "lib/peg.d.ts",
   "bin": "bin/peggy",
   "scripts": {
-    "pretest": "npm run parser",
-    "test": "npm run test:bare",
-    "test:bare": "mocha --recursive test --exclude test/browser.stub.js --reporter min",
+    "clean": "rimraf -f build && rimraf -f browser && mkdir build && cd build && mkdir ts && mkdir rollup && mkdir terser && cd .. && mkdir browser",
+    "terser": "terser -c passes=2 -m --sequences 40 build/rollup/peg.umd.js > build/terser/peggy.umd.min.js && terser -c passes=2 -m --sequences 40 --module build/rollup/peg.esmodule.js > build/terser/peggy.esmodule.min.js",
+    "rollup": "rollup -c",
+    "ts": "tsc --build tsconfig.json",
+    "set_version": "node -e \"const fs = require('fs'), pkg = JSON.parse(fs.readFileSync('./package.json')); fs.writeFileSync('./lib/version.js', '\\n\\'use strict\\';\\n\\nmodule.exports = \\'' + pkg.version + '\\';\\n');\"",
+    "test": "jest --coverage",
     "lint": "eslint . --ext .js bin/peggy",
-    "benchmark": "./benchmark/run",
-    "build": "npm run browser:build && npm run benchmark:build && npm run test:build",
-    "browser:clean": "rm -rf browser/",
-    "prebrowser:build": "npm run parser",
-    "browser:build": "browserify lib/peg.js -s peggy -o browser/peggy.min.js",
-    "postbrowser:build": "node tools/header browser/peggy.min.js && cp browser/peggy.min.js docs/vendor/peggy/peggy.min.js",
-    "benchmark:build": "browserify benchmark/browser.stub.js -o docs/js/benchmark-bundle.min.js && node tools/header docs/js/benchmark-bundle.min.js",
-    "test:build": "browserify test/browser.stub.js -o docs/js/test-bundle.min.js && node tools/header docs/js/test-bundle.min.js",
-    "parser": "node bin/peggy -o lib/parser.js --format commonjs src/parser.pegjs"
+    "benchmark": "node ./benchmark/run_bench.js",
+    "parser": "node bin/peggy -o lib/parser.js --format commonjs src/parser.pegjs",
+    "distro": "rm -f browser/peggy.min.js && cp build/terser/peggy.umd.min.js browser/peggy.min.js",
+    "audit": "npm run test && npm run benchmark",
+    "make": "npm run ts && npm run rollup && npm run terser && npm run parser",
+    "setup": "npm run clean && npm run set_version",
```

Comment:
I'm not sure that updating version (generally speaking, any activities that changes git repo status) is necessary here. That version number should be changed only explicitly on release

Reply:
> I'm not sure that updating version (generally speaking, any activities that changes git repo status) is necessary here. That version number should be changed only explicitly on release

[This was literally requested in a previous PR](https://github.com/peggyjs/peggy/issues/86) and was described as conceptually identical to what we already do.

I think it belongs here, because the concept here is that it has been moved to being part of the source, and as such, it needs to precede the build.

We have tests that enforce that the version number coming out of the library matches what's in package.  Those cannot succeed without this.

Reply:
It's really hard to maintain a library that pulls its own version into its source to be expressed at runtime without actually manufacturing that source

The major alternative is either to constantly JSON parse the package by hand directly in the source (gross, error prone,) or to try self-requiring, which has all sorts of weird consequences

This is simple, easy, and durable.  I hope we won't bikeshed it

Reply:
I don't think we're talking about how the file is created, just *when* the file is created.  Let's get to the point of having a design doc for the release process, and the answer will pop out of that.

Reply:
Also, if it were up to me, I'd remove `VERSION` from the API, but that's a breaking change.  We *could* change it to `please see package.json`...

Reply:
> I don't think we're talking about how the file is created, just when the file is created. 

If the version number is part of the library, then by definition it has to be before any build, or every build is out of date

I don't actually understand the purpose of the push back here.  What specific improvement is being sought?

&nbsp;

> Let's get to the point of having a design doc for the release process, and the answer will pop out of that.

😥 that's bikeshedding

&nbsp;

> Also, if it were up to me, I'd remove VERSION from the API

It's critically important to me.

&nbsp;

> We could change it to please see package.json...

The instinct to glad-handedly remove or change things just because they aren't personally important to you stands in direct contrast to the dedication you promised towards backwards compatability.

Engineering maturity is found in not requesting changes without a clearly stated reason.  Consequences are almost always much worse than you imagine.

Reply:
> If the version number is part of the library, then by definition it has to be before any build, or every build is out of date

Or, it's always changed when the version number is changed.

> I don't actually understand the purpose of the push back here. What specific improvement is being sought?

I want to understand the release process before I'll know if the build process adequately supports it.

> > Let's get to the point of having a design doc for the release process, and the answer will pop out of that.
> 
> 😥 that's bikeshedding

Design docs don't have to be complicated.  I'm thinking some bullet points in a markdown file that we can point people to when they want to understand how everything fits together.  We don't have to have that doc or understand that process before we land this patch, but I'd like to have it eventually.

> > Also, if it were up to me, I'd remove VERSION from the API
> 
> It's critically important to me.

Sorry, I thought you were arguing above that it should be removed.  I apologize for the misunderstanding.

> > We could change it to please see package.json...
> The instinct to glad-handedly remove or change things...

Nod.  I was saying that we *couldn't* change it, even thought it is generating all of this complexity.  I was unclear.

Reply:
> > If the version number is part of the library, then by definition it has to be before any build, or every build is out of date
>
> Or, it's always changed when the version number is changed.

This seems like what I'm doing.  I can't tell if I'm misunderstanding you or if we're coming to agreement

&nbsp;

> Sorry, I thought you were arguing above that it should be removed. I apologize for the misunderstanding.
>
> Nod. I was saying that we couldn't change it, even thought it is generating all of this complexity. I was unclear.

Okay, my mistake.

For clarity, I believe it cannot be removed.

For clarity, I believe my declaring `1.2.0` was a mistake.

What I'm defending is the mechanism, not that I had used it.

Reply:
> This seems like what I'm doing. I can't tell if I'm misunderstanding you or if we're coming to agreement

I think we're coming to an agreement, but I won't know for sure until I run your stuff myself.  I think we're getting really close, and I'm looking forward to getting to that.  I've got this PR queued up ahead of #117 and #116.